### PR TITLE
Tests for PBI Reports Direct Query and Large File Upload scenarios

### PR DESCRIPTION
**Pull request by parthsha**

### CODE REVIEW

File: src/RSAccessor/PortalAccessor.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        public PortalAccessor(string reportServerPortalUrl)
+        {
+            _reportServerApiUrl = reportServerPortalUrl + "/api/v2.0";
+        }
+
+        public ICredentials ExecuteCredentials { get; set; }
+
+        public HttpWebResponse DeleteFile(string path)
+        {
+            var query = string.Format("/CatalogItems(Path=%27{0}%27)", path); // TODO: path should be url encoded and single quotes should be doubled
```
Comment (lishibo) [inline src/RSAccessor/PortalAccessor.cs]: remove TODO?

File: src/RSAccessor/PortalAccessor.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            webRequest.Method = "DELETE";
+            webRequest.Credentials = ExecuteCredentials ?? CredentialCache.DefaultNetworkCredentials;
+            webRequest.Timeout = 1000 * 1000;
+
+            return (HttpWebResponse)webRequest.GetResponse();
+        }
+
+        public HttpWebResponse UploadLargeFile(string file, string type, string path)
+        {
+            var query = string.Format("/{0}s(Path=%27{1}%27)/Model.Upload", type, path); // TODO: path should be url encoded and single quotes should be doubled
```
Comment (lishibo) [inline src/RSAccessor/PortalAccessor.cs]: Same

File: src/RSLoad/MixedLoadWithPBI.loadtest [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+        <TestProfile Name="UsePowerBIReportsEmbeddedReuseModel" Path="bin\debug\rsload.dll" Id="78953d39-c69d-0c9e-2601-1fef9ea8926a" Percentage="34" Type="Microsoft.VisualStudio.TestTools.TestTypes.Unit.UnitTestElement, Microsoft.VisualStudio.QualityTools.Tips.UnitTest.ObjectModel, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" />
+        <TestProfile Name="UsePowerBIReportsEmbeddedStreamNewModel" Path="bin\debug\rsload.dll" Id="5d79ed0b-4372-43e2-bfc2-30387c73a685" Percentage="33" Type="Microsoft.VisualStudio.TestTools.TestTypes.Unit.UnitTestElement, Microsoft.VisualStudio.QualityTools.Tips.UnitTest.ObjectModel, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" />
+        <TestProfile Name="UseLiveConnectPBIReports" Path="bin\debug\rsload.dll" Id="41d1eef1-ef30-57f1-3342-5a9ccfe9132f" Percentage="33" Type="Microsoft.VisualStudio.TestTools.TestTypes.Unit.UnitTestElement, Microsoft.VisualStudio.QualityTools.Tips.UnitTest.ObjectModel, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" />
```
Comment (lishibo) [inline src/RSLoad/MixedLoadWithPBI.loadtest]: Add DirectQuery?

File: src/RSLoad/PowerBILoad.loadtest [hunk 4]
```diff
@@ -0,0 +1,1662 @@
...
+  <Scenarios>
+    <Scenario Name="PowerBI_Reports" DelayBetweenIterations="0" PercentNewUsers="0" IPSwitching="false" TestMixType="PercentageOfTestsStarted" ApplyDistributionToPacingDelay="true" MaxTestIterations="0" DisableDuringWarmup="false" DelayStartTime="0" AllowedAgents="">
+      <ThinkProfile Value="0.2" Pattern="NormalDistribution" />
+      <LoadProfile Pattern="Constant" InitialUsers="60" />
+      <TestMix>
+        <TestProfile Name="UsePowerBIReportsEmbeddedReuseModel" Path="bin\debug\rsload.dll" Id="78953d39-c69d-0c9e-2601-1fef9ea8926a" Percentage="25" Type="Microsoft.VisualStudio.TestTools.TestTypes.Unit.UnitTestElement, Microsoft.VisualStudio.QualityTools.Tips.UnitTest.ObjectModel, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" />
+        <TestProfile Name="UsePowerBIReportsEmbeddedStreamNewModel" Path="bin\debug\rsload.dll" Id="5d79ed0b-4372-43e2-bfc2-30387c73a685" Percentage="25" Type="Microsoft.VisualStudio.TestTools.TestTypes.Unit.UnitTestElement, Microsoft.VisualStudio.QualityTools.Tips.UnitTest.ObjectModel, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" />
```
Comment (lishibo) [inline src/RSLoad/PowerBILoad.loadtest]: We probably want to reduce the weight of "UsePowerBIReportsEmbeddedStreamNewModel" to 5.

**Review by lishibo — APPROVED** [review]
Also, could you update README about Power BI report tests? things like what tests are added, why weighted down "UsePowerBIReportsEmbeddedStreamNewModel", how to test with Scheduled data refresh, etc.

File: appveyor.yml [hunk 5]
```diff
@@ -5,8 +5,8 @@ before_build:
 before_test:
 - ps: |
     $ErrorActionPreference = 'Stop';
-    $sqlServerUrl = "$($env:downloadUrl)/SQLServer2016-x64-ENU.exe"
-    $sqlServerUrlBox = "$($env:downloadUrl)/SQLServer2016-x64-ENU.box"
+    $sqlServerUrl = "$($env:downloadUrl)/SQLServer2017-x64-ENU.exe"
```
Comment (jtarquino) [inline appveyor.yml]: You should merge from master where this have been updated

**Pull request merged**